airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py

```diff
@@ -42,6 +43,13 @@ class Config:
         examples=["2017-01-25T00:00:00Z"],
     )
 
+    end_date: Optional[datetime] = Field(
+        description="The end date until which you'd like to replicate data for AdCreatives and AdInsights APIs, in the format YYYY-MM-DDT00:00:00Z. All data generated between start_date and this date will be replicated.",
+        pattern="^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
+        examples=["2017-01-26T00:00:00Z"],
+        default=None,
+    )
```

Comment:
@keu The change was to made this Field as Optional and use default None, after set end_date to now.

Reply:
https://github.com/samuelcolvin/pydantic/pull/1210 please use inst

Reply:
having problem adding `default_factory=pendulum.now` can you help here @keu ?

Reply:
```
iff --git a/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py b/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py
index 386446175..b9a8857c8 100644
--- a/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py
+++ b/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py
@@ -47,7 +47,7 @@ class ConnectorConfig(BaseModel):
         description="The end date until which you'd like to replicate data for AdCreatives and AdInsights APIs, in the format YYYY-MM-DDT00:00:00Z. All data generated between start_date and this date will be replicated.",
         pattern="^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
         examples=["2017-01-26T00:00:00Z"],
-        default=None,
+        default_factory=pendulum.now,
     )

     include_deleted: bool = Field(default=False, description="Include data from deleted campaigns, ads, and adsets.")
```
works for me
```
start_date=datetime.datetime(2021, 5, 23, 0, 0, tzinfo=datetime.timezone.utc) 
end_date=DateTime(2021, 10, 5, 17, 11, 13, 63974, tzinfo=Timezone('Etc/UTC'))
```